### PR TITLE
some missing functions for OOK support in SX127X and RF69

### PR DESCRIPTION
**Pull request by cpainchaud**

### CODE REVIEW

File: src/modules/RF69/RF69.cpp [hunk 1]
```diff
@@ -369,6 +369,31 @@ int16_t RF69::readData(uint8_t* data, size_t len) {
...
+    }
+    if(state == ERR_NONE) {
+        _ook = enableOOK;
+    }
+
+    return(state);
+}
+
+int16_t RF69::setOokThresholdType(uint8_t type) {
+  if(type != RF69_OOK_THRESH_FIXED && type != RF69_OOK_THRESH_PEAK && RF69_OOK_THRESH_AVERAGE)
```
Comment (jgromes) [inline src/modules/RF69/RF69.cpp:388]: This doesn't seem correct, I think the last part should be `&& type != RF69_OOK_THRESH_AVERAGE` - please use parentheses to make the intent clear.

File: src/modules/RF69/RF69.cpp [hunk 2]
```diff
@@ -762,6 +787,13 @@ int16_t RF69::setEncoding(uint8_t encoding) {
   }
 }
 
+int16_t RF69::setLnaTestBoost(bool value) {
```
Comment (jgromes) [inline src/modules/RF69/RF69.cpp:790]: New public methods should have a keyword definition in keywords.txt (make sure to use true tabs and not spaces in that file).

File: src/modules/RF69/RF69.cpp [hunk 3]
```diff
@@ -762,6 +787,13 @@ int16_t RF69::setEncoding(uint8_t encoding) {
...
+  if(value)
```
Comment (jgromes) [inline src/modules/RF69/RF69.cpp:791]: Please use brackets even for single-line branches. It makes it easier to extend them later on.

File: src/modules/SX127x/SX1278.cpp [hunk 4]
```diff
@@ -315,8 +315,10 @@ int16_t SX1278::setOutputPower(int8_t power) {
 }
 
 int16_t SX1278::setGain(uint8_t gain) {
+    int16_t modem = getActiveModem();
+
   // check active modem
-  if(getActiveModem() != SX127X_LORA) {
+  if(modem != SX127X_LORA && modem != SX127X_FSK_OOK) {
```
Comment (jgromes) [inline src/modules/SX127x/SX1278.cpp:321]: This check is redundant now, the modem type is determined by a single bit, so there's only two options.

File: src/modules/SX127x/SX1278.h [hunk 5]
```diff
@@ -248,9 +248,11 @@ class SX1278: public SX127x {
     /*!
       \brief Gets recorded signal strength indicator of the latest received packet for LoRa modem, or current RSSI level for FSK modem.
 
+      \param skip_activation in OOK/FSK mode this function will put receiver un receive mode and in standby. Make it TRUE if you don't want this behaviour.
+
       \returns Last packet RSSI for LoRa modem, or current RSSI level for FSK modem.
     */
-    float getRSSI();
+    float getRSSI(bool skip_activation=false);
```
Comment (jgromes) [inline src/modules/SX127x/SX1278.h:255]: Please reflect the changes made in SX1278.h/SX1278.cpp in SX1272.h/SX1272.cpp too.

File: src/modules/SX127x/SX127x.cpp [hunk 6]
```diff
@@ -836,6 +836,44 @@ int16_t SX127x::disableAddressFiltering() {
...
+    return(ERR_WRONG_MODEM);
+  }
+
+  int16_t state = ERR_NONE;
+  state = _mod->SPIsetRegValue(SX127X_REG_OOK_FIX, value, 7, 0, 5);
+
+  return(state);
+}
+
+int16_t SX127x::setOokPeakThresholdDecrement(short int value) {
```
Comment (jgromes) [inline src/modules/SX127x/SX127x.cpp:863]: Please use stdint.h types (e.g. uint8_t instead of short int).

File: src/modules/SX127x/SX127x.cpp [hunk 7]
```diff
@@ -836,6 +836,44 @@ int16_t SX127x::disableAddressFiltering() {
...
+    return(ERR_WRONG_MODEM);
+  }
+
+  int16_t state = ERR_NONE;
+  state = _mod->SPIsetRegValue(SX127X_REG_OOK_PEAK, type, 4, 3, 5);
+
+  return(state);
+}
+
+int16_t SX127x::setOokFixedOrFloorThreshold(short int value) {
```
Comment (jgromes) [inline src/modules/SX127x/SX127x.cpp:851]: Please use stdint.h types (e.g. uint8_t instead of short int).

**Review by jgromes — CHANGES_REQUESTED** [review]
Hey, thanks for the PR - I left a couple comments, nothing major, mostly just style. Is there a specific reason for these additions (e.g. compatibility with existing hardware)?

File: src/modules/RF69/RF69.cpp (hunk 1 above)
Comment (cpainchaud) [reply src/modules/RF69/RF69.cpp:388]: > Hey, thanks for the PR - I left a couple comments, nothing major, mostly just style. Is there a specific reason for these additions (e.g. compatibility with existing hardware)?

Hi! My project mostly does OOK reception and your great library has a robust core but has simply put OOK on the side !

**Comment by cpainchaud** [comment]
> Hey, thanks for the PR - I left a couple comments, nothing major, mostly just style. Is there a specific reason for these additions (e.g. compatibility with existing hardware)?

Hi! My project mostly does OOK reception and your great library has a robust core but has simply put OOK on the side !

File: src/modules/RF69/RF69.cpp (hunk 1 above)
Comment (cpainchaud) [reply src/modules/RF69/RF69.cpp:388]: good catch!!!!

File: src/modules/RF69/RF69.cpp (hunk 3 above)
Comment (cpainchaud) [reply src/modules/RF69/RF69.cpp:791]: gotcha (and hopefully fixed now?)

File: src/modules/SX127x/SX1278.cpp (hunk 4 above)
Comment (cpainchaud) [reply src/modules/SX127x/SX1278.cpp:321]: perfect, removed!

File: src/modules/SX127x/SX1278.h (hunk 5 above)
Comment (cpainchaud) [reply src/modules/SX127x/SX1278.h:255]: done

File: src/modules/SX127x/SX127x.cpp (hunk 6 above)
Comment (cpainchaud) [reply src/modules/SX127x/SX127x.cpp:863]: ok

File: src/modules/SX127x/SX127x.cpp (hunk 7 above)
Comment (cpainchaud) [reply src/modules/SX127x/SX127x.cpp:851]: ok

**Review by jgromes — APPROVED** [review]

**Comment by jgromes** [comment]
Merged, thanks!

You're right, I haven't used OOK that much, so it's support isn't great. But that's why this is open soruce, so others may extend it too :)

**Pull request merged**